processing/src/main/java/io/druid/query/aggregation/cardinality/CardinalityAggregatorFactory.java

```diff
@@ -206,13 +262,13 @@ public String getName()
   @Override
   public List<String> requiredFields()
   {
-    return fieldNames;
+    return makeFieldNamesFromFields(fields);
```

Comment:
This isn't right; `requiredFields` should be the names of fields that this aggregator is going to read, but makeFIeldNamesFromFields returns the output names. I think switching that to getDimension instead of getOutputName would work.